lib/yard/cli/yardopts_command.rb

```diff
@@ -65,11 +65,14 @@ def yardopts_options(opts)
 
       private
 
-      # Parses the .yardopts file for default yard options
+      # Parses the .yardopts file for default yard options,
+      # ignoring comment lines beginning with // or #
       # @return [Array<String>] an array of options parsed from .yardopts
       def yardopts(file = options_file)
         return [] unless use_yardopts_file
-        File.read_binary(file).shell_split
+        contents = File.read_binary(file)
+        opts = contents.gsub(%r{(#|//).*\n?}, "")
```

Comment:
I'm completely for the constraint that this only support comments starting at the beginning of the line.

I'm happy to add more tests -- wanted to get your temperature on this first.